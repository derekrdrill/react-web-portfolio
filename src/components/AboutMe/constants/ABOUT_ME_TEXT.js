export const ABOUT_ME_TEXT = [
  {
    id: 'paragraph1',
    text: `Simple, but creative. Calm, but witty. Passionate, but measured.`,
  },
  {
    id: 'paragraph2',
    description: 'Paragraph 1',
    text: `Hi, I'm Derek. I'm just a guy from a small town in Maryland, 
        who now resides in the beautiful city of Charlotte.
        I remember falling in love with writing code, in an Intro to VB.Net class
        at Salisbury University. It all clicked for me. I was building the cheesiest
        'Movie Night' WinForms app you've ever seen, and it all 
        just started to make sense. I felt this fiery sensation
        inside as I built feature after feature, and solved problem after problem.
        The rush was addictive, and for someone who's always valued themself in working hard,
        solving problems and helping others, this was the high for me. I knew I wanted
        this to be my career.`,
  },
  {
    id: 'paragraph3',
    text: `After undergrad, I accepted a role as a level 1 programmer,
        working for a very small IT team, assigned to maintaining 
        and building onto their ERP software. This was very much an introduction
        to what it was like to be a professional developer, and was perfect
        for me to organically build a strong foundation of development principles.
        However, I knew I had to continue to sharpen my skills in my freetime.
        By taking online course after course, I continued to advance my skills
        in technologies like HTML, CSS, JavaScript, JQuery, Bootstrap and SQL,
        but also teaching myself more modern technologies and libraries, such as,
        Knockout.js, Node.js, React JS and MongoDB. I've carried those hard-working
        values, that I enstilled early on, through my career. 5 years in, and I am now
        a Front End Developer, and have accomplished things like:`,
  },
  {
    id: 'accolade1',
    text: `- Being apart of a team that entirely 
        rebuilt a legacy, desktop ERP application
        into a powerful modern web-app`,
  },
  {
    id: 'accolade2',
    text: `- Worked with clients at academic offices
        of education institutions to completely 
        automate their paper-processes by building
        workflows that comprised of powerful 
        web forms that integrated with their live ERP data`,
  },
  {
    id: 'accolade3',
    text: `- Heavily contributed towards the development of 
        a CDN library of reusable Knockout.js components
        that made the development of powerful web forms 
        (mentioned above) easier and less time consuming`,
  },
  {
    id: 'accolade4',
    text: `- Work as a React developer on a dev team 
        that is building and maintaining the IMPLAN App,
        a powerful economic analysis software tool
        `,
  },
  {
    id: 'paragraph4',
    text: `The last bullet is my current role, a Front End developer
        for the IMPLAN Group, and even after several years of hard work and continual education
        to get to this point, I continue to push the limits of my knowledge daily, in and out of the office, 
        to continue to grow as a professional front end developer`,
  },
  {
    id: 'paragraph5',
    text: `TLDR;`,
  },
  {
    id: 'paragraph6',
    text: `I love to write code because it fulfills the burning desire within me
        to work hard and solve intricate problems for others, and as an admitted nerd who 
        is not the most artistic, it's the best avenue to flexing my creativity muscles üí™üèº`,
  },
];
