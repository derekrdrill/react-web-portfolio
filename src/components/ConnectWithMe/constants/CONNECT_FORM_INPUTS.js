export const CONNECT_FORM_INPUTS = [
  {
    id: 1,
    title: 'Basic Info Inputs',
    inputs: [
      {
        id: 1,
        label: 'First Name',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
      {
        id: 2,
        label: 'Last Name',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
      {
        id: 3,
        label: 'Email',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
      {
        id: 4,
        label: 'Phone (Optional)',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
    ],
  },
  {
    id: 2,
    title: 'Message Text Area',
    inputs: [
      {
        id: 1,
        label: `Tell me somethin' good`,
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        multiline: true,
        minRows: 11,
        maxRows: 11,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
    ],
  },
];
