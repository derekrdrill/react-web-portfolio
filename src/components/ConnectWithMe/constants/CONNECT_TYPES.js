import React from 'react';
import { faEnvelopeSquare } from '@fortawesome/fontawesome-free-solid';
import { faGithubAlt, faLinkedin } from '@fortawesome/free-brands-svg-icons';

export const CONNECT_TYPES = [
  {
    id: 1,
    title: 'Email',
    icon: faEnvelopeSquare,
    href: 'mailto:derekrdrill@gmail.com',
    target: '',
    labelLeft: 'Email me',
  },
  {
    id: 2,
    title: 'LinkedIn',
    icon: faLinkedin,
    href: 'https://www.linkedin.com/in/derek-drill',
    target: '_blank',
    labelRight: 'Connect with my LinkedIn',
  },
  {
    id: 3,
    title: 'Github',
    icon: faGithubAlt,
    href: 'https://github.com/derekrdrill',
    target: '_blank',
    labelLeft: 'See this project on Github',
  },
];
