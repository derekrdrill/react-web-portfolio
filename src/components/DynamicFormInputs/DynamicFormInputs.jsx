import React, { useState } from 'react';
import styled from 'styled-components';
import { Grid, TextField, Select, MenuItem, FormControl, InputLabel } from '@mui/material';
import { ShowHideIcon } from '../ShowHideIcon/ShowHideIcon';

// TODO: condense this all to one style using emotion

// import { css } from '@emotion/react';

const FormInput = ({ input, onChange }) => {
  const [selectValue, setSelectValue] = useState('');
  const [passwordHidden, setPasswordHidden] = useState(true);
  const selectChange = event => setSelectValue(event.target.value);
  const handleSetPasswordHidden = () => setPasswordHidden(!passwordHidden);

  return (
    <Grid item xs={input.xs} sm={input.sm} md={input.md}>
      {input.select ? (
        <FormControl fullWidth>
          <InputLabel>{input.label}</InputLabel>
          <StyledSelect
            id={input.id}
            label={input.label}
            value={selectValue}
            variant={input.variant}
            fullWidth={input.fullWidth || true}
            onChange={selectChange}
            color={input.color || 'secondary'}
            inputbackgroundcolor={input.inputBackgroundColor}
            inputborderradius={input.inputborderRadius}
            inputpaddingbottom={input.inputPaddingBottom}
            inputfontfamily={input.fontFamily || input.inputFontFamily}
            labelcolor={input.labelColor}
            labelfontfamily={input.fontFamily || input.labelFontFamily}
          >
            {input.selectOptions.map(option => (
              <MenuItem key={option.id} value={option.value}>
                {option.value}
              </MenuItem>
            ))}
          </StyledSelect>
        </FormControl>
      ) : (
        <StyledTextField
          id={input.id}
          label={input.label}
          variant={input.variant}
          fullWidth={input.fullWidth || true}
          multiline={input.multiline}
          minRows={input.multiline && input.minRows}
          maxRows={input.multiline && input.maxRows}
          onChange={onChange}
          size={input.size || 'medium'}
          type={input.type === 'password' ? (passwordHidden ? 'password' : 'text') : input.type}
          color={input.color || 'secondary'}
          inputbackgroundcolor={input.inputBackgroundColor}
          inputborderradius={input.inputborderRadius}
          inputpaddingbottom={input.inputPaddingBottom}
          inputfontfamily={input.fontFamily || input.inputFontFamily}
          labelcolor={input.labelColor}
          labelfontfamily={input.fontFamily || input.labelFontFamily}
          InputProps={{
            endAdornment: input.type === 'password' && <ShowHideIcon onClick={handleSetPasswordHidden} />,
          }}
        />
      )}
    </Grid>
  );
};

export const DynamicFormInputs = ({ inputs, onChange }) => (
  <Grid container spacing={2}>
    {inputs.map(input => (
      <FormInput key={input.id} input={input} onChange={onChange} />
    ))}
  </Grid>
);

// TODO: condense this all to one style using emotion

// const styledInput = css(({
//     inputbackgroundcolor,
//     inputborderradius,
//     inputpaddingbottom,
//     inputfontfamily,
//     labelfontfamily,
//     labelcolor
// }) => ({
//     paddingBottom: inputpaddingbottom,
//     '.MuiInputBase-root': {
//         backgroundColor: inputbackgroundcolor,
//         borderRadius: inputborderradius,
//         fontFamily: inputfontfamily,
//     },
//     '.MuiInputLabel-root': {
//         color: labelcolor,
//         fontFamily: labelfontfamily,
//     },
// }));

const StyledTextField = styled(TextField)(
  ({ inputbackgroundcolor, inputborderradius, inputpaddingbottom, inputfontfamily, labelfontfamily, labelcolor }) => ({
    paddingBottom: inputpaddingbottom,
    '.MuiInputBase-root': {
      backgroundColor: inputbackgroundcolor,
      borderRadius: inputborderradius,
      fontFamily: inputfontfamily,
    },
    '.MuiInputLabel-root': {
      color: labelcolor,
      fontFamily: labelfontfamily,
    },
  }),
);

const StyledSelect = styled(Select)(
  ({ inputbackgroundcolor, inputborderradius, inputpaddingbottom, inputfontfamily, labelfontfamily, labelcolor }) => ({
    paddingBottom: inputpaddingbottom,
    '.MuiInputBase-root': {
      backgroundColor: inputbackgroundcolor,
      borderRadius: inputborderradius,
      fontFamily: inputfontfamily,
    },
    '.MuiInputLabel-root': {
      color: labelcolor,
      fontFamily: labelfontfamily,
    },
  }),
);
