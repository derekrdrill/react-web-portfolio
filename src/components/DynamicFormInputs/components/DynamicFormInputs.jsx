import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@mui/material';

import { FormInput } from './FormInput';

export const DynamicFormInputs = ({ inputs, form, setForm }) => (
  <Grid container spacing={2}>
    {inputs.map(input => {
      let value = form[input.id];

      return <FormInput key={input.id} form={form} input={input} setForm={setForm} value={value} />;
    })}
  </Grid>
);

DynamicFormInputs.propTypes = {
  inputs: PropTypes.array,
  form: PropTypes.object,
  setForm: PropTypes.func,
};
