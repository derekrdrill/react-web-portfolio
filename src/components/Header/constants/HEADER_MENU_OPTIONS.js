import React from 'react';
import AppsIcon from '@mui/icons-material/Apps';
import ConnectWithoutContactIcon from '@mui/icons-material/ConnectWithoutContact';
import PetsIcon from '@mui/icons-material/Pets';
import FaceIcon from '@mui/icons-material/Face';

export const HEADER_MENU_OPTIONS = [
  {
    id: 1,
    menuTitle: 'About Me',
    menuIcon: <FaceIcon />,
    mainTo: 'about-me',
    secondaryTo: '/about-me',
  },
  {
    id: 2,
    menuTitle: 'Projects',
    menuIcon: <AppsIcon />,
    mainTo: 'projects',
    secondaryTo: '/projects',
  },
  {
    id: 3,
    menuTitle: 'Connect With Me',
    menuIcon: <ConnectWithoutContactIcon />,
    mainTo: 'connect-with-me',
    secondaryTo: '/connect-with-me',
  },
  {
    id: 4,
    menuTitle: 'George',
    menuIcon: <PetsIcon />,
    mainTo: 'home',
    secondaryTo: '/',
  },
];
