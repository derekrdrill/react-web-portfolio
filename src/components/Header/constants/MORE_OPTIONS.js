import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEnvelopeSquare, faFile, faCode } from '@fortawesome/fontawesome-free-solid';
import { faGithubAlt, faLinkedin } from '@fortawesome/free-brands-svg-icons';

export const MORE_OPTIONS = [
  {
    id: 1,
    title: 'Email',
    icon: <FontAwesomeIcon icon={faEnvelopeSquare} />,
    href: 'mailto:derekrdrill@gmail.com',
    target: '',
  },
  {
    id: 2,
    title: 'LinkedIn',
    icon: <FontAwesomeIcon icon={faLinkedin} />,
    href: 'https://www.linkedin.com/in/derek-drill',
    target: '_blank',
  },
  {
    id: 3,
    title: 'Github',
    icon: <FontAwesomeIcon icon={faGithubAlt} />,
    href: 'https://github.com/derekrdrill',
    target: '_blank',
    divider: true,
  },
  {
    id: 4,
    title: 'Resume',
    icon: <FontAwesomeIcon icon={faFile} />,
    target: '',
    href: '',
    modal: {
      modalTitle: 'Select one',
      modalDescription: 'Would you like to download a copy or be sent to the Google doc?',
      modalButton1: 'Download .DOCX',
      modalButton2: 'Go to Google Doc',
      modalCloseButton: 'Cancel',
    },
  },
  {
    id: 5,
    title: 'Source Code',
    icon: <FontAwesomeIcon icon={faCode} />,
    target: '',
    href: '',
  },
];
