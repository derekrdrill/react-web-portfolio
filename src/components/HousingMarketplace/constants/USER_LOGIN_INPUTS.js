export const USER_LOGIN_INPUTS = [
  {
    id: 'username',
    label: 'Username',
    variant: 'outlined',
    xs: 12,
    fullWidth: true,
  },
  {
    id: 'password',
    label: 'Password',
    variant: 'outlined',
    xs: 12,
    fullWidth: true,
    type: 'password',
  },
];
