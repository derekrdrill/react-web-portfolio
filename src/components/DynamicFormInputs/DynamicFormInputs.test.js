import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { DynamicFormInputs } from './DynamicFormInputs';
import { CONNECT_FORM_INPUTS } from '../../components/ConnectWithMe/constants/CONNECT_FORM_INPUTS';

const renderer = new ShallowRenderer();

describe('Connect with me tests', () => {
  it('renders correctly', () => {
    renderer.render(<DynamicFormInputs inputs={CONNECT_FORM_INPUTS[0].inputs} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
