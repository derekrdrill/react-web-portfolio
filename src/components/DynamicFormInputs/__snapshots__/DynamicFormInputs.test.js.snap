// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect with me tests renders correctly 1`] = `
<ForwardRef(Grid)
  container={true}
  spacing={2}
>
  <FormInput
    input={
      Object {
        "fontFamily": "Kufam",
        "fullWidth": true,
        "id": 1,
        "inputBackgroundColor": "gainsboro",
        "inputPaddingBottom": 5,
        "inputborderRadius": 4,
        "label": "First Name",
        "labelColor": "#28282B",
        "variant": "filled",
        "xs": 12,
      }
    }
  />
  <FormInput
    input={
      Object {
        "fontFamily": "Kufam",
        "fullWidth": true,
        "id": 2,
        "inputBackgroundColor": "gainsboro",
        "inputPaddingBottom": 5,
        "inputborderRadius": 4,
        "label": "Last Name",
        "labelColor": "#28282B",
        "variant": "filled",
        "xs": 12,
      }
    }
  />
  <FormInput
    input={
      Object {
        "fontFamily": "Kufam",
        "fullWidth": true,
        "id": 3,
        "inputBackgroundColor": "gainsboro",
        "inputPaddingBottom": 5,
        "inputborderRadius": 4,
        "label": "Email",
        "labelColor": "#28282B",
        "variant": "filled",
        "xs": 12,
      }
    }
  />
  <FormInput
    input={
      Object {
        "fontFamily": "Kufam",
        "fullWidth": true,
        "id": 4,
        "inputBackgroundColor": "gainsboro",
        "inputPaddingBottom": 5,
        "inputborderRadius": 4,
        "label": "Phone (Optional)",
        "labelColor": "#28282B",
        "variant": "filled",
        "xs": 12,
      }
    }
  />
</ForwardRef(Grid)>
`;
